R/format_PFN.R

```diff
@@ -0,0 +1,589 @@
+                  .data$CapturePopID, .data$CapturePlot, .data$ReleasePopID, .data$ReleasePlot,
+                  .data$Mass, .data$Tarsus, .data$OriginalTarsusMethod, .data$WingLength,
+                  .data$Age_observed, .data$Age_calculated, .data$ExperimentID,
+                  .data$BroodID)
+
+  ## 11) Return data
+  return(Capture_data)
+}
+
+#' Create individual data table for EastDartmoor.
+#'
+#' @param Capture_data Capture data table generated by create_capture_EDM
+#'
+#' @return A data frame with Individual data
+
+create_individual_EDM <- function(Capture_data){
+
+  #Take capture data and determine summary data for each individual
+  Indv_data <- Capture_data %>%
+    dplyr::arrange(.data$IndvID, .data$BreedingSeason, .data$CaptureDate, .data$CaptureTime) %>%
+    dplyr::group_by(.data$IndvID) %>%
+
+    #* CN: Loop over the species assigned to each individual and check the number of different assignments
+
+    dplyr::summarise(Species = purrr::map_chr(.x = list(unique(na.omit(.data$Species))), .f = ~{
+
+      if(length(..1) == 0){
+
+        return(NA_character_)
+
+      } else if(length(..1) == 1){
+
+        return(..1)
+
+      } else {
+
+        return("CCCCCC")
+
+      }
+
+    }), PopID = "EDM",
+    BroodIDLaid = first(.data$BroodID),
+    BroodIDFledged = .data$BroodIDLaid, # Identical, as no cross-fostering experiments were made
+    RingSeason = first(.data$BreedingSeason),
+    RingAge = ifelse(any(.data$Age_calculated %in% c(1, 3)), "chick", ifelse(min(.data$Age_calculated) == 2, NA_character_, "adult")),
+    Sex_calculated = purrr::map_chr(.x = list(unique(na.omit(.data$Sex_observed))), .f = ~{
+
+      if(length(..1) == 0){
+
+        return(NA_character_)
+
+      } else if(length(..1) == 1){
+
+        return(..1)
+
+      } else {
+
+        return("C")
+
+      }
+
+    })) %>%
+    dplyr::rowwise() %>%
+
+
+    #For each individual, if their ring age was 1 or 3 (caught in first breeding year)
+    #Then we take their first BroodID, otherwise it is NA
+    dplyr::mutate(BroodIDLaid = ifelse(.data$RingAge == "chick", .data$BroodIDLaid, NA),
+                  BroodIDFledged = .data$BroodIDLaid) %>%
+    #Ungroup to prevent warnings in debug report
+    dplyr::ungroup() %>%
+
+    # Add a column for genetic sex (not available here)
+    dplyr::mutate(Sex_genetic = NA_character_) %>%
+
+    # Sort
+    dplyr::arrange(.data$RingSeason, .data$IndvID)
+
+  return(Indv_data)
+}
+
+#' Create location data table for EastDartmoor.
+#'
+#' @param Brood_data Brood data table generated by create_brood_EDM
+#' @param Capture_data Capture data table generated by create_capture_EDM
+#'
+#' @return A data frame with Location data
+
+create_location_EDM <- function(Brood_data, Capture_data){
+
+  # 1) Load and format additional data on nest boxes
+  Location_details <- utils::read.csv(file = paste0(db, "/PFN_PrimaryData_EDartmoor_Nestboxes.csv"), na.strings = c("", "?"),colClasses = "character") %>%
```

Comment:
This will fail because the 'db' object doesn't exist inside the function. Maybe better to load this data already on l. 70

Reply:
I also get some problems with the column names on Windows. Specifying the fileEncoding should fix this (and hopefully doesn't cause any problems on Mac).
```suggestion
  Location_details <- utils::read.csv(file = paste0(db, "/PFN_PrimaryData_EDartmoor_Nestboxes.csv"), na.strings = c("", "?"),colClasses = "character", fileEncoding="UTF-8-BOM") %>%
```

Reply:
I am now loading the data in the beginning.